internal/setup/setup_test.go

```diff
@@ -1063,6 +1063,179 @@ var setupTests = []setupTest{{
 		`,
 	},
 	relerror: `invalid slice definition filename: "a.yaml"`,
+}, {
+	summary: "Package essentials with same package slice",
+	input: map[string]string{
+		"slices/mydir/mypkg.yaml": `
+			package: mypkg
+			essential:
+				- mypkg_slice2
+			slices:
+				slice1:
+				slice2:
+				slice3:
+					essential:
+						- mypkg_slice1
+						- mypkg_slice4
+				slice4:
+					essential:
+						- mypkg_slice2
+		`,
+	},
+	release: &setup.Release{
+		DefaultArchive: "ubuntu",
+		Archives: map[string]*setup.Archive{
+			"ubuntu": {
+				Name:       "ubuntu",
+				Version:    "22.04",
+				Suites:     []string{"jammy"},
+				Components: []string{"main", "universe"},
+				PubKeys:    []*packet.PublicKey{testKey.PubKey},
+			},
+		},
+		Packages: map[string]*setup.Package{
+			"mypkg": {
+				Archive: "ubuntu",
+				Name:    "mypkg",
+				Path:    "slices/mydir/mypkg.yaml",
+				Slices: map[string]*setup.Slice{
+					"slice1": {
+						Package: "mypkg",
+						Name:    "slice1",
+						Essential: []setup.SliceKey{
+							{"mypkg", "slice2"},
+						},
+					},
+					"slice2": {
+						Package: "mypkg",
+						Name:    "slice2",
+					},
+					"slice3": {
+						Package: "mypkg",
+						Name:    "slice3",
+						Essential: []setup.SliceKey{
+							{"mypkg", "slice2"},
+							{"mypkg", "slice1"},
+							{"mypkg", "slice4"},
+						},
+					},
+					"slice4": {
+						Package: "mypkg",
+						Name:    "slice4",
+						Essential: []setup.SliceKey{
+							{"mypkg", "slice2"},
+							{"mypkg", "slice2"},
```

Comment:
[Comment for reviewer] This is existing behavior. We can change it to be a set and throw an error if a slice is present twice in an essentials list.